lib/posix/posix.nim

```diff
@@ -564,7 +564,8 @@ proc pread*(a1: cint, a2: pointer, a3: int, a4: Off): int {.
 proc pwrite*(a1: cint, a2: pointer, a3: int, a4: Off): int {.
   importc, header: "<unistd.h>".}
 proc read*(a1: cint, a2: pointer, a3: int): int {.importc, header: "<unistd.h>".}
-proc readlink*(a1, a2: cstring, a3: int): int {.importc, header: "<unistd.h>".}
+when not defined(nintendoswitch):
+  proc readlink*(a1, a2: cstring, a3: int): int {.importc, header: "<unistd.h>".}
```

Comment:
`symlink` is defined in the Nintendo Switch SDK, but not `readlink`?

Reply:
it's just newlib. For Switch specifically it shouldn't matter either way though, because the OS only supports SD cards formatted with FAT32 or exFAT and to my knowledge neither support symlinks.

Reply:
Well, I've explicitly revoked readlink, as I cannot efficiently stub it. It must break the code that tries to use it, so the developer will be aware of it not working, as it may be hard to debug if stubbed.

But as for the symlink creation, the non-working part is easy to see (nothing gets into some dir), so I made it result in a "nothing done" behaviour:
```
when not defined(nintendoswitch):
  proc symlink*(a1, a2: cstring): cint {.importc, header: "<unistd.h>".}
else:
  proc symlink*(a1, a2: cstring): cint = -1
```

Reply:
So, could you please tell me what to fix in the way I used or resolve this conversation? :)